src/verifiable/credential.ts

```diff
@@ -134,6 +137,128 @@ export class W3CCredential {
     return undefined;
   }
 
+  /**
+   * Verify credential proof
+   *
+   * @returns {*}  {(boolean)}
+   */
+  async verifyProof(
+    proofType: ProofType,
+    resolverURL: string,
+    credStatusResolverRegistry?: CredentialStatusResolverRegistry
+  ): Promise<boolean> {
+    const proof = this.getProofByType(proofType);
+    if (!proof) {
+      throw new Error('proof not found');
+    }
+
+    const coreClaim = this.getCoreClaimFromProof(proofType);
+    if (!coreClaim) {
+      throw new Error(`can't get core claim`);
+    }
+
+    switch (proofType) {
+      case ProofType.BJJSignature: {
+        if (!credStatusResolverRegistry) {
+          throw new Error('please provide credential status resolver registry');
+        }
+        const userDID = DID.parse(this.credentialSubject.id as string);
```

Comment:
theoretically, it could be self credential and the id will be missing. please, use the function [get](https://github.com/0xPolygonID/js-sdk/blob/d9aabff5d05c94d93196f2f43561e412aef362b1/src/credentials/utils.ts#L13)